src/System/extract_system_rep.rs

```diff
@@ -168,3 +195,48 @@ pub fn get_system_recipe(
         _ => panic!("Got unexpected query side: {:?}", side),
     }
 }
+
+fn validate_reachability_input(
+    machine: &SystemRecipe,
+    start: &QueryExpression,
+    end: &QueryExpression,
+) {
+    let mut components: usize = 0;
+    count_component(machine, &mut components);
```

Comment:
Why not make `count_component` just return a `usize`? Seems kinda redundant to have an output parameter. `components` could also be immutable then

Reply:
Fixed